src/overlays/actors/ovl_En_Slime/z_en_slime.c

```diff
@@ -45,146 +73,1195 @@ ActorInit En_Slime_InitVars = {
+            velocity.y = (Rand_ZeroOne() * 6.0f) + 2.0f;
+            EffectSsDtBubble_SpawnCustomColor(play, &effectPos, &velocity, &sBubbleAccel, &sBubblePrimColor,
+                                              &sBubbleEnvColor, Rand_S16Offset(40, 20), 20, 0);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2F140.s")
+        if (this->actor.params == EN_SLIME_TYPE_YELLOW) {
+            Item_DropCollectible(play, &this->actor.world.pos, ITEM00_ARROWS_10);
+        } else if (this->actor.params == EN_SLIME_TYPE_GREEN) {
+            Item_DropCollectible(play, &this->actor.world.pos, ITEM00_MAGIC_SMALL);
+        } else if (this->actor.params == EN_SLIME_TYPE_RED) {
+            Item_DropCollectible(play, &this->actor.world.pos, ITEM00_RECOVERY_HEART);
+        }
+        EnSlime_SetupWaitForRevive(this);
+    }
+    this->actor.scale.y = this->actor.scale.x;
+    this->actor.scale.z = this->actor.scale.x;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2F180.s")
+/**
+ * Calculate a new "current" position value that is a multiple of 60
+ * away from the "home" position.
+ * This is for use by blue chuchus frozen into an ice block so that
+ * their position is a multiple of the ice block size.
+ */
+f32 EnSlime_SnapIceBlockPosition(f32 currentPosition, f32 homePosition) {
+    s32 dist = currentPosition - homePosition;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2F1A4.s")
+    if (dist > 0) {
+        dist += 30;
+    } else {
+        dist -= 30;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2F354.s")
+    return ((dist / 60) * 60.0f) + homePosition;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2F418.s")
+/**
+ * Prepare actor for ice block to spawn around it.
+ * The actor will stop moving, have AC_ON cleared,
+ * and target position will be snapped to
+ * a multiple of the size of an ice block.
+ */
+void EnSlime_SetupSpawnIceBlock(EnSlime* this) {
+    this->collider.base.acFlags &= ~AC_ON;
+    this->actor.flags &= ~ACTOR_FLAG_1;
+    this->drawDmgEffAlpha = 0.0f;
+    this->actor.speed = 0.0f;
+    this->actor.velocity.y = 0.0f;
+    this->actor.gravity = 0.0f;
+    this->iceBlockSnapPos.x = EnSlime_SnapIceBlockPosition(this->actor.world.pos.x, this->actor.home.pos.x);
+    this->iceBlockSnapPos.z = EnSlime_SnapIceBlockPosition(this->actor.world.pos.z, this->actor.home.pos.z);
+    Actor_SetColorFilter(&this->actor, 0x8000, 0x80FF, 0x2000, 10);
+    this->eyeTexIndex = EN_SLIME_EYETEX_OPEN;
+    this->iceBlockTimer = 254;
+    this->actionFunc = EnSlime_SpawnIceBlock;
+}
+
+/**
+ * Attempt to move the chuchu to snapped position and
+ * spawn an ice block actor as a child of the chuchu actor.
+ * If ice block spawn fails, chuchu returns to idle.
+ */
+void EnSlime_SpawnIceBlock(EnSlime* this, PlayState* play) {
+    s32 stepCheck; // bool
```

Comment:
I don't think you need to say this is a bool - there are hundreds, if not thousands of s32s in the codebase used like bools. It may be unfamiliar to people new to the codebase, but it's a *very* established pattern.